pkg/cluster/driver/capi/capi.go

```diff
@@ -224,6 +204,98 @@ func (cad *ClusterApiDriver) getWorkloadClusterApplications(restConfig *rest.Con
 	return ret, nil
 }
 
+func (cad *ClusterApiDriver) getOciCcmOptions(restConfig *rest.Config) error {
+	// If the values are already set, don't try to set them.  This accounts
+	// for two cases: this fuction has already been called, or there are
+	// specific values set in the cluster configuration.
+	if cad.ClusterConfig.Providers.Oci.Vcn != "" && cad.ClusterConfig.Providers.Oci.LoadBalancer.Subnet1 != "" && cad.ClusterConfig.Providers.Oci.LoadBalancer.Subnet2 != "" {
+		return nil
+	}
+
+	// The values are not populated.  Go get the OCICluster associated
+	// with the Cluster and find the values.
+	ociCluster, err := cad.getOCIClusterObject()
+	if err != nil {
+		return err
+	}
+
+	ociClusterNs := ociCluster.GetNamespace()
+	ociClusterName := ociCluster.GetName()
+
+	err = k8s.GetResource(restConfig, &ociCluster)
+	if err != nil {
+		return err
+	}
+
+
+	// The values that are required are buried inside .spec.networkSpec.vcn
+	spec, err := getMapVal(ociCluster.Object, "spec", ociClusterNs, ociClusterName)
+	if err != nil {
+		return err
+	}
+
+	networkSpec, err := getMapVal(spec, "networkSpec", ociClusterNs, ociClusterName)
+	if err != nil {
+		return err
+	}
+
+	vcn, err := getMapVal(networkSpec, "vcn", ociClusterNs, ociClusterName)
+	if err != nil {
+		return err
+	}
+
+	vcnId, err := getStringVal(vcn, "id", ociClusterNs, ociClusterName)
+	if err != nil {
+		return err
+	}
+	log.Debugf("Found VCN OCID %s", vcnId)
+
+	if vcnId == "" {
```

Comment:
Is the user supposed to set the VCN OCID in this case?  Otherwise, would it be the controller and it is this something that might take time to get set (e.i. race condition).  Same for subnets.